src/components/ExperimentalFeaturesMenu.tsx

```diff
@@ -0,0 +1,84 @@
+import { Popover } from '@headlessui/react'
+import toast from 'react-hot-toast'
+
+import { DEFAULT_DEFAULT_EXPERIMENTAL_FEATURES } from '@src/lib/constants'
+import { kclManager } from '@src/lib/singletons'
+import { err, reportRejection } from '@src/lib/trap'
+import { useEffect, useState } from 'react'
+import { CustomIcon } from '@src/components/CustomIcon'
+import { warningLevels } from '@src/lib/settings/settingsTypes'
+import type { WarningLevel } from '@rust/kcl-lib/bindings/WarningLevel'
+import { setExperimentalFeatures } from '@src/lib/kclHelpers'
+
+export function ExperimentalFeaturesMenu() {
+  const [fileSettings, setFileSettings] = useState(kclManager.fileSettings)
+
+  const currentLevel: WarningLevel =
+    fileSettings.experimentalFeatures ?? DEFAULT_DEFAULT_EXPERIMENTAL_FEATURES
+
+  useEffect(() => {
+    setFileSettings(kclManager.fileSettings)
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- TODO: blanket-ignored fix me!
+  }, [kclManager.fileSettings])
+
+  return (
+    currentLevel.type !== 'Deny' && (
```

Comment:
This component is copied from UnitsMenu, biggest difference is we're hiding it if experimentalFeatures are off. I don't think people should necessarily discover this on their own outside of an experimental command